python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -548,16 +564,21 @@ def _get_mirror_list(self, repo, url):
                 )
                 msg = msg.replace(url, repl_url)
                 log(0, msg)
-            # no mirror list found continue without
+            # no mirror list or metalink found continue without
             return returnlist
 
         def _replace_and_check_url(url_list):
             goodurls = []
             skipped = None
             for url in url_list:
-                # obvious bogons get ignored b/c, we could get more interesting checks but <shrug>
+                # obvious bogons get ignored
                 if url in ['', None]:
                     continue
+                # Discard any urls containing some invalid characters
+                forbidden_characters = "<>^`{|}"
```

Comment:
Is it still required as we are not reading HTML as a list of URLs anymore?

Reply:
I think they could still upload a plaintext list of urls with invalid entries containing forbidden characters, so now this checks for that. Probably not a common problem but I figured it couldn't hurt to keep the check.

Reply:
Do we have examples of repo URLs containing such characters?

Reply:
I've not seen any. (I based the list of forbidden characters on the response here: https://stackoverflow.com/a/36667242 . If we think there are people using urls that actually contain those characters somehow, I'll take it out again.)

Reply:
Hmmm, if we include this extra check, I would make some log/warning message to be displayed in case we are excluding an URL as invalid.

Reply:
Good point, I'll add something for that.